src/compiler/checker.ts

```diff
@@ -14588,6 +14592,48 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
             type.constructSignatures = constructSignatures;
         }
+
+        // And lastly, if the underlying symbol has an export=, we need to copy the type structure from it

```

Comment:
This logic and the `getExportsOfModuleWorker` should be identical, except the former is only concerned with copying values for the sake of symbolic traversal, while this copies up all type structure from the `export=` target - members, signatures, and index infos.